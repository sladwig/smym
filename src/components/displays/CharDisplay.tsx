import React from 'react';
import { PositionEmitter } from '../PositionEmitter';

export const CharDisplay = ({ char }: { char: { position: number; value: string } }) => {
    return <PositionEmitter position={char.position}>{char.value}</PositionEmitter>;
};
