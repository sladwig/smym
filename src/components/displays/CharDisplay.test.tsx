import { render } from '@testing-library/react';
import React from 'react';
import { CharDisplay } from './CharDisplay';

describe('CharDisplay', () => {
    test('renders', () => {
        const { container } = render(<CharDisplay char={{ position: 1, value: '' }} />);
        expect(container).toMatchSnapshot();
    });
});
