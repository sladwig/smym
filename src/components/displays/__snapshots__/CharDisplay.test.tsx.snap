// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharDisplay renders 1`] = `
<div>
  <span
    class="position-emitter"
  >
    
    <span
      class="position-area-overlay"
    >
      <span
        class="position-area"
      />
      <span
        class="position-area"
      />
    </span>
  </span>
</div>
`;
